src/Service/Doctrine/ResultProvider.php

```diff
@@ -317,7 +318,31 @@ private function calculateTotalCount(Pager $filter, int $resultCount)
     private function findCount(AnalysedQuery $analysedQuery): int
     {
         $countQueryBuilder = $analysedQuery->cloneQueryBuilder();
+        $groupByParts = $countQueryBuilder->getDQLPart('groupBy');
         $countQueryBuilder->select(sprintf('count(%s)', $analysedQuery->getRootAlias()));
+
+        $this->validateGroupByParts($groupByParts);
+        if (count($groupByParts) === 1) {
+            return count($countQueryBuilder->getQuery()->getArrayResult());
```

Comment:
Could be extremely slow in some situations. Changing to `SELECT COUNT(DISTINCT groupByField)` and removing group by query would be much better.

Also, if you're just counting like you do, no point in not allowing multiple group by columns, as this would also work.